### PR TITLE
Add DistSQL to set Proxy props.

### PR DESCRIPTION
**Pull request by lanchengx**
Fixed https://github.com/apache/shardingsphere/issues/13207

Changes proposed in this pull request:
- `set variable statement` supports configuration property
- Add tests.
- Improve documentation.

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/executor/SetVariableExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+public final class SetVariableExecutorTest {
+    
+    @Test
+    public void assertExecuteWithTransactionType() {
+        SetVariableStatement statement = new SetVariableStatement("transaction_type", "local");
+        BackendConnection connection = mock(BackendConnection.class);
+        when(connection.getTransactionStatus()).thenReturn(new TransactionStatus(TransactionType.XA));
+        new SetVariableExecutor(statement, connection).execute();
+        Assert.assertEquals(TransactionType.LOCAL.name(), connection.getTransactionStatus().getTransactionType().name());
```
Comment (TeslaCN) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/executor/SetVariableExecutorTest.java:42]: Consider using `assertThat` here and below.

Comment (lanchengx) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/executor/SetVariableExecutorTest.java]: thanks, I have corrected it

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/excutor/SetVariableExecutor.java [hunk 2]
```diff
@@ -41,20 +51,57 @@
...
+    private Object getValue(final ConfigurationPropertyKey propertyKey, final String value) {
+        try {
+            TypedPropertyValue propertyValue = new TypedPropertyValue(propertyKey, value);
+            return propertyValue.getValue();
+        } catch (TypedPropertyValueException e) {
+            throw new InvalidValueException(value);
+        }
+    }
+    
+    private void handleVariables(final SetVariableStatement statement) {
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/excutor/SetVariableExecutor.java:92]: The variable name `statement` represents a lot of meaning, can it be more precise?

**Comment by menghaoranss** [comment]
@lanchengx CI build failed.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/excutor/SetVariableExecutor.java [hunk 3]
```diff
@@ -41,20 +51,57 @@
...
+        if (metaDataPersistService.isPresent() && null != metaDataPersistService.get().getPropsService()) {
+            metaDataPersistService.get().getPropsService().persist(props, true);
+        }
+    }
+    
+    private Object getValue(final ConfigurationPropertyKey propertyKey, final String value) {
+        try {
+            TypedPropertyValue propertyValue = new TypedPropertyValue(propertyKey, value);
+            return propertyValue.getValue();
+        } catch (TypedPropertyValueException e) {
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/excutor/SetVariableExecutor.java:87]: ex is better.

**Comment by lanchengx** [comment]
> @lanchengx CI build failed.

I have corrected it

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**